pkg/testingsdk/rde_utils.go

```diff
@@ -50,11 +66,88 @@ func GetComponentMapInStatus(ctx context.Context, client *vcdsdk.Client, cluster
 }
 
 func GetKubeconfigFromRDEId(ctx context.Context, client *vcdsdk.Client, clusterId string) (string, error) {
-	capvcdStatusMap, err := GetComponentMapInStatus(ctx, client, clusterId, vcdsdk.ComponentCAPVCD)
+	var kubeConfig string
+	var err error
+
+	clusterRde, err := getRdeById(ctx, client, clusterId)
 	if err != nil {
-		return "", fmt.Errorf("error retrieving [%s] field in status field of RDE [%s]: [%v]", vcdsdk.ComponentCAPVCD, clusterId, err)
+		return "", fmt.Errorf("error occurred while getting cluster RDE [%s]: [%v]", clusterId, err)
+	}
+
+	entityTypeSemver, _ := semver.New(getRDEVersion(clusterRde))
+	entityTypeVersion120, _ := semver.New(CAPVCDEntityTypeVersion120)
+
+	// If RDE version of the cluster is >= 1.2.0, the kubeconfig will be stored as a secure field within the RDE's CAPVCD status
+	// else, the kubeconfig can be directly obtained from the RDE
+	if entityTypeSemver.GE(*entityTypeVersion120) {
+		kubeConfig, err = getKubeconfigFromDecryptedRDE(ctx, client, clusterId, clusterRde.Name)
+	} else {
+		kubeConfig, err = getKubeconfigFromRDE110(ctx, client, clusterId)
+	}
+
+	if err != nil {
+		return "", fmt.Errorf("error occurred retrieving kubeconfig for cluster [%s(%s)] from RDE: [%v]", clusterRde.Name, clusterId, err)
+	}
+
+	if kubeConfig == "" {
+		return "", fmt.Errorf("error occurred, empty kubeconfig was retrieved from cluster RDE [%s(%s)]", clusterRde.Name, clusterId)
+	}
+	return kubeConfig, err
+}
+
+func getDecryptedRDE(ctx context.Context, client *vcdsdk.Client, rdeID, clusterName string) (*FullCAPVCDEntity, error) {
```

Comment:
I think we should be moving this code to the common core as we are using the same code in CSE too. We could do this later on as well.

Reply:
Sure, I agree with that as well. Let me create a ticket for this to move it over to common-core, given the time constraint we have I think we can do this after GA as we will probably need to re-arrange some structs. Thanks Aniruddha!

Reply:
Created a ticket as VKCI-197